### PR TITLE
Balder/add multibitvector iterator too conformance tests too

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe PR
@geirst and @toregge FYI

### CODE REVIEW

File: searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp [hunk 1]
```diff
@@ -506,6 +510,57 @@ Test::testEndGuard(bool invert)
...
+        }
+    }
+}
+Verifier::~Verifier() = default;
+
+SearchIterator::UP
+Verifier::create(bool strict) const {
+    MultiSearch::Children bvs;
+    for (const auto & bv : _bvs) {
+        bvs.push_back(BitVectorIterator::create(bv.get(), getDocIdLimit(), _tfmd, strict, false).release());
```
Comment (havardpe) [inline searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp]: in the blueprint code the 'strict' parameter is inherited differently for AND(first child) and OR(all children). This is minor and should not impact the actual testing.

**Review by havardpe — APPROVED** [review]
consider converting test to new syntax

File: searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp [hunk 2]
```diff
@@ -506,6 +510,57 @@ Test::testEndGuard(bool invert)
...
+    }
+    SearchIterator::UP iter(_is_and ? AndSearch::create(bvs, strict) : OrSearch::create(bvs, strict));
+    return MultiBitVectorIteratorBase::optimize(std::move(iter));
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp]: Fixed

File: searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp]: Correct. So add a test that strictness is as expected.

File: searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp [hunk 3]
```diff
@@ -551,6 +551,7 @@ Verifier::create(bool strict) const {
     SearchIterator::UP iter(_is_and ? AndSearch::create(bvs, strict) : OrSearch::create(bvs, strict));
     auto mbvit = MultiBitVectorIteratorBase::optimize(std::move(iter));
     EXPECT_TRUE((bvs.size() < 2) || (dynamic_cast<const MultiBitVectorIteratorBase *>(mbvit.get()) != nullptr));
+    EXPECT_EQUAL(strict, Trinary::True == mbvit->is_strict());
```
Comment (havardpe) [inline searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp]: this is also good to verify, but it was not the strictness I was talking about (and it should be implicitly verified by the conformance test as well).

I was talking about the fact that non-first children to AND are *not strict* in production, but *are* strict in this test (before they are optimized away).

Comment (baldersheim) [reply searchlib/src/tests/queryeval/multibitvectoriterator/multibitvectoriterator_test.cpp]: Yes, I am aware of that. But as you say. That does not matter here. I just added a sanity check here to make it clear what we were testing.

**Pull request merged**